third_party/txt/src/skia/paragraph_builder_skia.cc

```diff
@@ -112,10 +168,14 @@ skt::TextStyle TxtToSkia(const TextStyle& txt) {
 
   skia.setLocale(SkString(txt.locale.c_str()));
   if (txt.has_background) {
-    skia.setBackgroundColor(txt.background);
+    skia.setBackgroundPaintID(CreatePaintID(txt.background_dl));
   }
   if (txt.has_foreground) {
-    skia.setForegroundColor(txt.foreground);
+    skia.setForegroundPaintID(CreatePaintID(txt.foreground_dl));
+  } else {
+    flutter::DlPaint dl_paint;
```

Comment:
This piece is new and I'm not sure why it's needed now...?

Reply:
The TextStyle API has `color` and `foreground` fields that are somewhat redundant.  `color` sets the color of the text, and `foreground` sets all of the painting properties of the text including the color.  If `foreground` is set, then it takes precedence over `color`.

Before this PR the style's `color` field would be forwarded to its SkPargraph counterpart.  Now, if `color` is used then Flutter must create a `DlPaint` representing the color.